qiime2/core/type/primitive.py

```diff
@@ -188,6 +188,10 @@ def to_ast(self):
 
 class Choices(Predicate):
     def __init__(self, choices):
+
+        if not choices:
+            raise ValueError("Choices must have a minimum of one value.")
```

Comment:
This is a pretty reasonable error, but maybe make it a little more mechanical in this case as our audience here are plugin developers (who are more or less familiar with Python).
```
'Choices' cannot be instantiated with an empty set.
```
The single-quotes are commonly used in Python exceptions to disambiguate words from actual objects/variables.